### PR TITLE
Fix watchpoints ("memory breakpoints") with JIT

### PR DESCRIPTION
**Pull request by comex**
They didn't work at all except in the interpreter.

Also fix some trivial GUI issues that broke the Dolphin debugger on (at least) OS X.

This seems to work for a basic watchpoint in Melee, but it could use much more testing.

**Update: fixed the below issue.  Haven't addressed Lioncash's comments yet.**

:warning: I'm encountering a serious problem with the debugger which I haven't pinned down yet, but I can reproduce it on master, so it shouldn't block this PR.  Since the Dolphin project does not have a usable bug tracker, I will describe it here:

When (a) I'm using a debug build, (b) the debugger is enabled, and (c) fastmem is on (note: this PR disables fastmem if there are any watchpoints), attempting to boot Melee ends up producing this wonderful noise:

https://dl.dropboxusercontent.com/u/1237941/bzzzzzzz.m4a (warning: loud)

and these OSReports:

```
assertion "cobj" failed in cobj.c on line 709.
DATE Feb 13 2002  TIME 22:06:27
- STACK ---------------------------------------------
 Address:  Back Chain  LR Save
804EE9E0:   804EEA08   803882EC
804EEA08:   804EEA28   80388260
804EEA28:   804EEA48   803686DC
804EEA48:   804EEAD0   803685A0
804EEAD0:   804EEAE8   803910F4
804EEAE8:   804EEAF8   8039101C
804EEAF8:   804EEB48   801A5048
804EEB48:   804EEB80   801A40F4
804EEB80:   804EEBB0   801A44C4
804EEBB0:   804EEBD0   801A45C8
804EEBD0:   804EEBF8   801601AC
804EEBF8:   FFFFFFFF   8000533C
```

### CODE REVIEW

File: Source/Core/Common/BreakPoints.h [hunk 1]
```diff
@@ -105,6 +106,8 @@ class MemChecks
 	void Remove(u32 _Address);
 
 	void Clear() { m_MemChecks.clear(); }
+
+	bool HasAny() { return !m_MemChecks.empty(); }
```
Comment (lioncash) [inline Source/Core/Common/BreakPoints.h]: this can be const

File: Source/Core/DolphinWX/Debugger/MemoryWindow.cpp [hunk 2]
```diff
@@ -94,7 +94,13 @@ CMemoryWindow::CMemoryWindow(wxWindow* parent, wxWindowID id,
 	sizerBig->Add(sizerRight, 0, wxEXPAND | wxALL, 3);
 	sizerRight->Add(addrbox = new wxTextCtrl(this, IDM_MEM_ADDRBOX, ""));
 	sizerRight->Add(valbox = new wxTextCtrl(this, IDM_VALBOX, ""));
-	sizerRight->Add(new wxButton(this, IDM_SETVALBUTTON, _("Set &Value")));
+	sizerRight->Add(new wxButton(this, IDM_SETVALBUTTON,
+#ifdef __APPLE__
+		_("Set Value")
+#else
+		_("Set &Value")
+#endif
```
Comment (lioncash) [inline Source/Core/DolphinWX/Debugger/MemoryWindow.cpp]: Why is this necessary?

Comment (comex) [inline Source/Core/DolphinWX/Debugger/MemoryWindow.cpp]: See the commit message for the commit that introduced this.

Comment (lioncash) [inline Source/Core/DolphinWX/Debugger/MemoryWindow.cpp]: I'd opt for just removing the accelerator. Having an accelerator there is very out of place.

Comment (comex) [inline Source/Core/DolphinWX/Debugger/MemoryWindow.cpp]: Okay.  There are also a bunch of other dubious accelerators - two options labeled `&D`, three `&U`.  I guess I should just remove all the accelerators in the file (reminder to self: see whether pressing enter on the value textbox triggers Set Value).

**Comment by comex** [comment]
Addressed Lioncash and magumagu's comments.

**Comment by skidau** [comment]
Nice work.  Sorely needed feature.
